packages/react/src/analytics/integrations/GA4/types/index.ts

```diff
@@ -0,0 +1,65 @@
+import type {
+  EventData,
+  IntegrationOptions,
+  LoadIntegrationEventData,
+  SetUserEventData,
+  trackTypes,
+  TrackTypesValues,
+} from '@farfetch/blackout-analytics';
+import type { Schema } from '../../shared/types/shared.types';
+
+export type Schemas = Record<string, Schema>;
+
+export interface GA4IntegrationOptions extends IntegrationOptions {
+  measurementId: string;
+  enableAutomaticPageViews?: boolean;
+  nonInteractionEvents?: NonInteractionEvents;
+  productMappings?: ProductMappings;
+  scopeCommands?: ScopeCommands;
+  schemas?: Schemas;
+  ScopeCommands?: boolean;
+  dataLayerName?: string;
+  loadScript?: () => Promise<void>;
+  onPreProcessCommands?: OnPreProcessCommandsHandler;
```

Comment:
Missing `setCustomUserIdProperty` option, which should be a `boolean`. Please, review this type and make sure that all the options are defined here. You can look at the `constants` module for all the variables that begin with `OPTION` to know all the options that are available here.